lib/cnf_graph.ml

```diff
@@ -305,3 +305,106 @@ let get_cnf g1 g2 max_time nbr_colors =
       assert (l1 = l2 && w1 = w2);
       []
 (*never return this list*)
+
+let pp_graph fmt g w l =
+  let rec aux_print_graph x y =
+    if y = l then ()
+    else if x = w then (
+      Format.fprintf fmt "\n";
+      aux_print_graph 0 (y + 1))
+    else (
+      Format.fprintf fmt "%d " g.(y).(x);
+      aux_print_graph (x + 1) y)
+  in
+  aux_print_graph 0 0
```

Comment:
I don’t know what is best. I know that for my previous tests I wrote this function with iter 🤷‍♂️:
```suggestion
let pp_graph graph =
      Array.iter
      (fun current_row ->
        Array.iter
          (fun cell ->
            print_int cell;
            print_string " ")
          current_row;
        print_newline ())
      grid;
```

Reply:
It’s just a suggestion. I don’t know if it’s really helpful 😂

Reply:
I can keep your part with the iter which is surely more readable, however I keep the Format.fprintf because otherwise my function is more of a pretty printer.